### ISSUE TITLE
Modals: No header example does not aria-hide the rest of the page

### ISSUE THREAD

**Issue by arodenbeck** [salesforce/design-system-react#1236]
The 2nd example, no header, does not appear to hide the rest of the page. This allows a screen reader user to perceive parts of the page that should not be functional.

**Comment by arodenbeck** [comment]
The aria-labeledby does not seem to be properly labeling  this modal either. I'm guessing the referenced Id should be in the header, which this variant does not have. An aria-label could be used here instead, or possibly pass in a label.
